koku/api/provider/serializers.py

```diff
@@ -67,6 +75,39 @@ class Meta:
         fields = ('uuid', 'provider_resource_name', 'credentials')
 
 
+class AWSAuthenticationSerializer(ProviderAuthenticationSerializer):
+    """AWS auth serializer."""
+
+
+class AzureAuthenticationSerializer(ProviderAuthenticationSerializer):
+    """Azure auth serializer."""
+
+    def validate_credentials(self, creds):
+        """Validate credentials field."""
+        key = 'provider.credentials'
+        fields = ['subscription_id', 'tenant_id', 'client_id', 'client_secret']
+        return validate_field(creds, fields, key)
+
+
+class GCPAuthenticationSerializer(ProviderAuthenticationSerializer):
+    """GCP auth serializer."""
+
+    def validate_credentials(self, creds):
+        """Validate credentials field."""
+        key = 'provider.credentials'
+        fields = ['project_id']
+        return validate_field(creds, fields, key)
+
+
+class OCPAuthenticationSerializer(ProviderAuthenticationSerializer):
+    """OCP auth serializer."""
+
+
+class OCPAWSAuthenticationSerializer(AWSAuthenticationSerializer,
```

Comment:
Shouldn't need this class. The source created is going to be either OCP or AWS.

Reply:
Agreed, it's a pass-through class. It's here for completeness and to help avoid unintentionally biasing the code toward any particular cloud provider.